### PR TITLE
API: add DataTransfer

### PR DESCRIPTION
**Pull request by caugner**
[DataTransfer](https://developer.mozilla.org/docs/Web/API/DataTransfer)

### CODE REVIEW

File: api/DataTransfer.json [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+            "version_added": false
+          }
+        },
+        "status": {
+          "experimental": true,
+          "standard_track": true,
+          "deprecated": false
+        }
+      },
+      "Constructor": {
```
Comment (connorshea) [inline api/DataTransfer.json]: This should be called DataTransfer as well, but with a description that specifies it's a constructor.

File: api/DataTransfer.json [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+              "version_added": null
+            }
+          },
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": true
+          }
+        },
+        "getData": {
```
Comment (connorshea) [inline api/DataTransfer.json]: `getData`, `setData`, and so on are subfeatures of `types` because of the hierarchy of the file I think.

**Comment by connorshea** [comment]
This is missing a few Firefox-specific properties: `mozCursor`, mozItemCount`, `mozSourceNode`, `mozUserCancelled` 

https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer#Properties

File: api/DataTransfer.json (hunk 1 above)
Comment (caugner) [reply api/DataTransfer.json]: Thanks, I have fixed this in d281f23.

File: api/DataTransfer.json (hunk 2 above)
Comment (caugner) [reply api/DataTransfer.json]: Sorry, I just fixed this in 7efdf4f.

**Comment by caugner** [comment]
@connorshea Thanks for the review, I have now added the moz* sub-features in 7811112.

File: api/DataTransfer.json [hunk 3]
```diff
@@ -0,0 +1,966 @@
...
+        "status": {
+          "experimental": true,
+          "standard_track": true,
+          "deprecated": false
+        }
+      },
+      "DataTransfer": {
+        "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/DataTransfer/DataTransfer",
+          "description": "<code>DataTransfer</code> constructor",
```
Comment (connorshea) [inline api/DataTransfer.json]: This should use `DataTransfer()`

Comment (caugner) [reply api/DataTransfer.json]: Thanks, I have just fixed that in fade90d (and rebased on upstream/master).

**Comment by caugner** [comment]
@connorshea I think this is ready. Would you mind taking another look?

File: api/DataTransfer.json [hunk 4]
```diff
@@ -0,0 +1,966 @@
...
+              "version_added": null
+            },
+            "samsunginternet_android": {
+              "version_added": null
+            }
+          },
+          "status": {
+            "experimental": true,
+            "standard_track": true,
+            "deprecated": true
```
Comment (connorshea) [inline api/DataTransfer.json]: I don't think getData is experimental or deprecated?

File: api/DataTransfer.json [hunk 5]
```diff
@@ -0,0 +1,966 @@
...
```
Comment (connorshea) [inline api/DataTransfer.json]: Neither is setData

File: api/DataTransfer.json [hunk 6]
```diff
@@ -0,0 +1,966 @@
...
```
Comment (connorshea) [inline api/DataTransfer.json]: or clearData

File: api/DataTransfer.json (hunk 4 above)
Comment (connorshea) [inline api/DataTransfer.json]: I don't think getData is experimental or deprecated?

File: api/DataTransfer.json (hunk 5 above)
Comment (connorshea) [inline api/DataTransfer.json]: Neither is setData

File: api/DataTransfer.json (hunk 6 above)
Comment (connorshea) [inline api/DataTransfer.json]: Neither is clearData

File: api/DataTransfer.json [hunk 7]
```diff
@@ -0,0 +1,966 @@
...
```
Comment (connorshea) [inline api/DataTransfer.json]: or files, it looks like you should re-review all of these statuses (unless I'm missing something?)

File: api/DataTransfer.json [hunk 8]
```diff
@@ -0,0 +1,966 @@
...
+              "version_added": false
+            },
+            "edge": {
+              "version_added": false
+            },
+            "edge_mobile": {
+              "version_added": false
+            },
+            "firefox": {
+              "version_added": true
```
Comment (connorshea) [inline api/DataTransfer.json]: The article says Chrome supports this and Firefox doesn't, which makes no sense to me. I assume you looked into this and came to the same conclusion, just figured I'd mention it.

Comment (caugner) [reply api/DataTransfer.json]: Yes, I have checked [in the Firefox repo](https://dxr.mozilla.org/mozilla-central/source/dom/events/DataTransfer.h), see [this commit](https://github.com/mdn/browser-compat-data/commit/38782d29f14f08613754165437e1f74bdb9db81a).

File: api/DataTransfer.json (hunk 4 above)
Comment (caugner) [reply api/DataTransfer.json]: Fixed in b8ea1d5 now. Thanks again!

File: api/DataTransfer.json (hunk 5 above)
Comment (caugner) [reply api/DataTransfer.json]: dito

File: api/DataTransfer.json (hunk 6 above)
Comment (caugner) [reply api/DataTransfer.json]: dito

File: api/DataTransfer.json (hunk 7 above)
Comment (caugner) [reply api/DataTransfer.json]: dito

**Pull request merged**